js_modules/dagit/packages/core/src/nav/useRepositoryLocationReload.tsx

```diff
@@ -56,52 +65,103 @@ export const useRepositoryLocationReload = (location: string) => {
 
   const invalidateConfigs = useInvalidateConfigsForRepo();
 
-  const tryReload = React.useCallback(async () => {
-    dispatch({type: 'start-reloading'});
-    const {data} = await reload();
-    dispatch({type: 'finish-reloading'});
+  const {startPolling, stopPolling} = useQuery<RepositoryLocationStatusQuery>(
+    REPOSITORY_LOCATION_STATUS_QUERY,
+    {
+      skip: state.pollStartTime === null,
+      pollInterval: 5000,
+      fetchPolicy: 'no-cache',
+      // This is irritating, but apparently necessary for now.
+      // https://github.com/apollographql/apollo-client/issues/5531
+      notifyOnNetworkStatusChange: true,
+      onCompleted: (data: RepositoryLocationStatusQuery) => {
+        const workspace = data.workspaceOrError;
+
+        // On any failure, immediately show the error dialog. This is a blocking failure that must be
+        // either retried after repairing the issue or dismissed manually.
+        if (workspace.__typename !== 'Workspace') {
```

Comment:
if it's a PythonError we should show the error stack trace like we do in other contexts